kroxylicious-kms-provider-hashicorp-vault-test-support/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/VaultKmsTestUtils.java

```diff
@@ -0,0 +1,164 @@
+
+package io.kroxylicious.kms.provider.hashicorp.vault;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.kroxylicious.kms.service.KmsException;
+import io.kroxylicious.kms.service.UnknownAliasException;
+
+/**
+ * The type Vault kms test utils.
+ */
+public class VaultKmsTestUtils {
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final HttpClient vaultClient = HttpClient.newHttpClient();
+    public static final String VAULT_ROOT_TOKEN = "rootToken";
+
+    private VaultKmsTestUtils() {
+    }
+
+    /**
+     * Create vault get http request.
+     *
+     * @param url the url
+     * @return the http request
+     */
+    public static HttpRequest createVaultGet(URI url) {
+        return createVaultRequest()
+                .uri(url)
+                .GET()
+                .build();
+    }
+
+    /**
+     * Create vault delete http request.
+     *
+     * @param url the url
+     * @return the http request
+     */
+    public static HttpRequest createVaultDelete(URI url) {
+        return createVaultRequest()
+                .uri(url)
+                .DELETE()
+                .build();
+    }
+
+    /**
+     * Create vault post http request.
+     *
+     * @param url the url
+     * @param bodyPublisher the body publisher
+     * @return the http request
+     */
+    public static HttpRequest createVaultPost(URI url, HttpRequest.BodyPublisher bodyPublisher) {
+        return createVaultRequest()
+                .uri(url)
+                .POST(bodyPublisher).build();
+    }
+
+    private static HttpRequest.Builder createVaultRequest() {
+        return HttpRequest.newBuilder()
+                .header("X-Vault-Token", VAULT_ROOT_TOKEN)
+                .header("Accept", "application/json");
+    }
+
+    /**
+     * Send request.
+     *
+     * @param <R>  the type parameter
+     * @param key the key
+     * @param request the request
+     * @param valueTypeRef the value type ref
+     * @return the typeRef
+     */
+    public static <R> R sendRequest(String key, HttpRequest request, TypeReference<R> valueTypeRef) {
+        try {
+            HttpResponse<byte[]> response = vaultClient.send(request, HttpResponse.BodyHandlers.ofByteArray());
+            if (response.statusCode() == 404) {
+                throw new UnknownAliasException(key);
+            }
+            else if (response.statusCode() != 200) {
+                throw new IllegalStateException("unexpected response %s for request: %s".formatted(response.statusCode(), request.uri()));
+            }
+            byte[] body = response.body();
+            return decodeJson(valueTypeRef, body);
```

Comment:
I'm aware I've paged out a *lot* of context about how we've ended up here, so forgive me if I'm re-hashing something we have talked about before. 

It feels strange that the test Kek Manager knows anything about HTTP status codes let alone what they mean in domain terms. I think thats telling us we have the abstraction wrong, we have our Kek Managers tired to tightly to the request clients they are managing. 

The Test Kek manager should just be providing extra functionality (as specified by the TestKekManager interface)  to the production Kek Manager it shouldn't be re-implementing behaviour. I'm not even convinced the `TestKekManager` needs to be injected into the production code, as it could just ensure an environment is correctly prepared for the tests...

Reply:
This is one of the questions that came to my mind, but I was afraid to remove too many things here. We are adding some error handling here and we want to remove it from the KEK manager.

We would need to return somehow the status code or the whole response and do something with that response in the test case.

What if we avoid all status code checking here (Vault and AWS Kek Managers) and move the checking to some test utils to decode json bodies, check status code of the response and so on as part of the tests?

Reply:
I would keep this PR focused on removing the check then act pattern.

Reply:
I think Keith's right lets keep this focused on the narrower refactoring.